capty.py

```
@@ -253,104 +950,152 @@ def on_record_clicked(self, btn):
+        self.record_btn.get_style_context().remove_class("record-button")
+        self.record_btn.get_style_context().add_class("stop-button")
+        self.status.set_text("Starting recording...")
 
         def record_worker():
             try:
-                # short delay
-                time.sleep(delay)
-                GLib.idle_add(self.status.set_text, "Recording...")
-                # Start ffmpeg as a process group so we can stop it gracefully
+                logger.info("Starting recording process")
                 self.ffproc = subprocess.Popen(ff_cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, preexec_fn=os.setsid)
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)